src/Syntax/Implicits.hs

```diff
@@ -284,6 +286,13 @@ matches (TyPromotedCon t) (TyPromotedCon t') = t == t'
 matches TyPromotedCon{} _ = False
 
 matches (TyApp f x) (TyApp f' x') = f `matches` f' && x `matches` x'
+
+matches (TyApp f x) (a :-> b) =
+  matches f (TyApp (TyCon (TgName (T.pack "->") (negate 7))) a) && matches x b
```

Comment:
This seems incredibly fragile. How are we doing this elsewhere?

Reply:
Exactly like this (but using the builtin name)

Reply:
(done)